File: src/brs/db/store/ATStore.java
package brs.db.store;

import brs.at.AT;
import brs.at.AT.AtMapEntry;
import brs.db.SignumKey;
import brs.db.VersionedEntityTable;
import brs.util.CollectionWithIndex;

import java.util.Collection;
import java.util.List;

public interface ATStore {

  boolean isATAccountId(Long id);

  List<Long> getOrderedATs();

  AT getAT(Long id);

  AT getAT(Long id, int height);

  Collection<AT> getATs(Collection<Long> ids);

  List<Long> getATsIssuedBy(Long accountId, Long codeHashId, int from, int to);

  Collection<Long> getAllATIds(Long codeHashId);

  SignumKey.LongKeyFactory<AT> getAtDbKeyFactory();

  VersionedEntityTable<AT> getAtTable();

  SignumKey.LongKeyFactory<AT.ATState> getAtStateDbKeyFactory();

  VersionedEntityTable<AT.ATState> getAtStateTable();

  VersionedEntityTable<brs.at.AT.AtMapEntry> getAtMapTable();

  Long findTransaction(int startHeight, int endHeight, Long atID, int numOfTx, long minAmount);

  int findTransactionHeight(Long transactionId, int height, Long atID, long minAmount);

  AtMapEntry getMapValueEntry(long atId, long key1, long key2);

  long getMapValue(long atId, long key1, long key2);

  CollectionWithIndex<AtMapEntry> getMapValues(long atId, long key1, Long value, int from, int to );
}

File: src/main/java/brs/db/store/ATStore.kt
package brs.db.store

import brs.at.AT
import brs.db.BurstKey
import brs.db.VersionedEntityTable

interface ATStore {
    suspend fun getOrderedATs(): List<Long>

    suspend fun getAllATIds(): Collection<Long>

    val atDbKeyFactory: BurstKey.LongKeyFactory<AT>

    val atTable: VersionedEntityTable<AT>

    val atStateDbKeyFactory: BurstKey.LongKeyFactory<AT.ATState>

    val atStateTable: VersionedEntityTable<AT.ATState>

    suspend fun isATAccountId(id: Long?): Boolean

    suspend fun getAT(id: Long?): AT?

    suspend fun getATsIssuedBy(accountId: Long?): List<Long>

    suspend fun findTransaction(startHeight: Int, endHeight: Int, atID: Long?, numOfTx: Int, minAmount: Long): Long?

    suspend fun findTransactionHeight(transactionId: Long?, height: Int, atID: Long?, minAmount: Long): Int
}
